src/api/controllers/reactions.ts

```diff
@@ -51,20 +51,22 @@ export async function handleInternalMessageReactionAdd(data: DiscordPayload) {
     guildID: payload.guild_id || "",
   };
 
-  eventHandlers.reactionAdd?.(
-    uncachedOptions,
-    payload.emoji,
-    payload.user_id,
-    message,
-  );
+  if (typeof payload.emoji === "string") {
+    eventHandlers.reactionAdd?.(
+      uncachedOptions,
+      payload.emoji,
+      payload.user_id,
+      message,
+    );
+  }
 }
 
 export async function handleInternalMessageReactionRemove(
-  data: DiscordPayload,
+  data: GatewayPayload,
 ) {
   if (data.t !== "MESSAGE_REACTION_REMOVE") return;
 
-  const payload = data.d as MessageReactionPayload;
+  const payload = data.d as MessageReactionAddEventPayload;
```

Comment:
The message reaction add event payload differs slightly in the documentation, so `MESSAGE_REACTION_REMOVE` should have its own interface. 

Reference: <https://discord.com/developers/docs/topics/gateway#message-reaction-remove-message-reaction-remove-event-fields>

Reply:
Good catch. Thats just a typo here